File: python/2610.Convert-an-Array-Into-a-2D-Array-With-Conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        result = []
        freq = defaultdict(int)

        for num in nums:
            freq[num] += 1
            row = freq[num] - 1

            if row == len(result):
                result.append([])

            result[row].append(num)

        return result

File: javascript/2610.Convert-an-Array-Into-a-2D-Array-With-Conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function (nums) {
  const result = [];
  const freq = new Map();

  for (let i = 0; i < nums.length; i++) {
    freq.set(nums[i], (freq.get(nums[i]) || 0) + 1);

    const row = freq.get(nums[i]) - 1;

    if (!result[row]) result[row] = [];
    result[row].push(nums[i]);
  }

  return result;
};
